transferui/src/main/java/com/hyperwallet/android/ui/transfer/view/ListTransferDestinationActivity.java

```diff
@@ -0,0 +1,134 @@
+
+    public static final String ARGUMENT_SELECTED_DESTINATION = "argument_selected_destination";
+    private static final String TAG = ListTransferDestinationActivity.class.getSimpleName();
+
+    private ListTransferDestinationViewModel mListTransferDestinationViewModel;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_list_transfer_destination);
+
+        if (savedInstanceState == null) {
+            initFragment(ListTransferDestinationFragment.newInstance(
+                    (HyperwalletTransferMethod) getIntent().getParcelableExtra(ARGUMENT_SELECTED_DESTINATION)));
+        }
+
+        mListTransferDestinationViewModel = ViewModelProviders.of(this, new ListTransferDestinationViewModel
+                .ListTransferDestinationViewModelFactory(
+                TransferMethodRepositoryFactory.getInstance().getTransferMethodRepository())).get(
+                ListTransferDestinationViewModel.class);
+
+        registerObservers();
+    }
+
+    @Override
+    public boolean onSupportNavigateUp() {
+        onBackPressed();
+        return true;
+    }
+
+    @Override
+    public void onBackPressed() {
+        getWindow().clearFlags(WindowManager.LayoutParams.FLAG_TRANSLUCENT_STATUS);
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_DRAWS_SYSTEM_BAR_BACKGROUNDS);
+        getWindow().setStatusBarColor(ContextCompat.getColor(this, R.color.colorPrimaryDark));
+        getWindow().getDecorView().setSystemUiVisibility(0);
+        super.onBackPressed();
+    }
+
+    private void initFragment(Fragment fragment) {
+        FragmentManager fragmentManager = getSupportFragmentManager();
+        FragmentTransaction transaction = fragmentManager.beginTransaction();
+        transaction.add(R.id.frame, fragment);
+        transaction.commit();
+    }
+
+    private void registerObservers() {
+        mListTransferDestinationViewModel.getTransferDestinationSection().observe(this,
+                new Observer<Event<HyperwalletTransferMethod>>() {
+                    @Override
+                    public void onChanged(Event<HyperwalletTransferMethod> destination) {
+                        Intent resultIntent = new Intent();
+                        resultIntent.putExtra(ARGUMENT_SELECTED_DESTINATION, destination.getContent());
+                        setResult(RESULT_OK, resultIntent);
+                        finish();
+                    }
+                });
+
+        mListTransferDestinationViewModel.getTransferDestinationError().observe(this,
+                new Observer<Event<HyperwalletErrors>>() {
+                    @Override
+                    public void onChanged(Event<HyperwalletErrors> errorsEvent) {
+                        showError(errorsEvent.getContent().getErrors());
```

Comment:
Please make sure to consume the event only once

Reply:
Sorry, I can't get this. Actually we consume an Event just ones.